### PR TITLE
ENG-6592 - Simplify build gradle & some functions

### PR DESCRIPTION
**Pull request by kevinsites**
I changed the build.gradle file for the app to use Flavor Dimensions/Variants (https://developer.android.com/build/build-variants) in order to simplify our code structure. Now instead of copying files that aren't relevant for the ADV libs (and having an Android ADV and RN ADV), we have one shared ADV folder that will be pulled in for the Android/RN ADV build respectively. Moving forward if we need to add any additional custom libraries we can use this pattern to share code across the Android/RN platforms.

Additionally I refactored the NIDIdentifyAllViews files in both the Android/RN directory to use shared functions/logic across the two platforms.

### CODE REVIEW

**Comment by kishisaka-nid** [comment]
The advanced unit tests (FPJSHelperTest) are failing. I think this is because of the removed flavor. 
`Task 'testAdvancedDeviceAndroidLibDebugUnitTest' not found in project ':NeuroID'.`
Have update the GitHub-actions-unit-test.yaml file and move that test into advancedDeviceLib.

**Comment by kishisaka-nid** [comment]
The sample app is not able to see the extension when we change the variant to androidAdvancedDeviceLibDebug. Cannot see the method start(true).

**Comment by kishisaka-nid** [comment]
All ok now after 2nd commit. 🚢 it!

**Review by kishisaka-nid — APPROVED** [review]

File: NeuroID/src/reactNativeLib/java/com/neuroid/tracker/callbacks/NIDFragmentCallbacks.kt
Comment (priya-xavier) [inline NeuroID/src/reactNativeLib/java/com/neuroid/tracker/callbacks/NIDFragmentCallbacks.kt:1]: The reactNativeLib folder didn't get deleted even though the files in it were.

**Review by cometman — APPROVED** [review]

**Review by priya-xavier — APPROVED** [review]
Looks great!

**Pull request merged**